src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -149,6 +155,126 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     }
                 }
 
+                var nuGetLockFilePath = PackagesLockFileUtilities.GetNuGetLockFilePath(_request.Project);
+                PackagesLockFile nuGetLockFile = null;
+
+                var restorePackagesWithLockFile = _request.Project.RestoreMetadata?.RestoreLockProperties.RestorePackagesWithLockFile;
+
+                if (!MSBuildStringUtility.IsTrueOrEmpty(restorePackagesWithLockFile) && File.Exists(nuGetLockFilePath))
+                {
+                    _success = false;
+
+                    // invalid input since packages.lock.json file exists along with RestorePackagesWithLockFile is set to false.
+                    var message = string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidLockFileInput, nuGetLockFilePath);
+                    await _logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1005, message));
+
+                    // Replay Warnings and Errors from an existing lock file
+                    await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(_request.ExistingLockFile, _logger);
+
+                    // Write the logs into the assets file
+                    var logs = _logger.Errors
+                        .Select(l => AssetsLogMessage.Create(l))
+                        .ToList();
+
+                    LockFile newAssetsFile = null;
+                    if (_request.ExistingLockFile != null)
+                    {
+                        newAssetsFile = _request.ExistingLockFile.Clone();
+                        newAssetsFile.LogMessages = logs;
```

Comment:
I'm confused...you replay the logs only to override them completely?

Reply:
it generates a new error, replays existing ones, and then add them all to new lock file. I'll try to make it more simple to understand or add comments.